test/functional/mining_basic.py

```diff
@@ -115,6 +116,46 @@ def test_blockmintxfee_parameter(self):
             assert tx_below_min_feerate['txid'] not in block_template_txids
             assert tx_below_min_feerate['txid'] not in block_txids
 
+    def test_timewarp(self):
+        self.log.info("Test timewarp attack mitigation (BIP94)")
+        node = self.nodes[0]
+
+        self.log.info("Mine until the last block of the retarget period")
+        blockchain_info = self.nodes[0].getblockchaininfo()
+        n = 144 - blockchain_info['blocks'] % 144 - 2
```

Comment:
It would be nice to reduce the magic numbers in this test a bit

`DIFFICULTY_ADJUSTMENT_INTERVAL` instead of 144, `MAX_FUTURE_BLOCK_TIME` instead of 2 * 3600, etc.

Reply:
I reduced the magic.